tools/CACertificates/CACertificateOverview.md

```diff
@@ -109,8 +109,8 @@ This will create files mydevice* that contain the public key, private key, and P
 
 #### IoT Leaf Device
 
-* Run `./certGen.sh create_device_certificate mydevice` to create the new device certificate.  
-  This will create the files ./certs/new-device.* that contain the public key and PFX and ./private/new-device.key.pem that contains the device's private key.  
+* Run `./certGen.sh create_device_certificate mydevice` to create the new device certificate. If the device certificate needs to be signed by intermediate CA, run `./certGen.sh create_device_certificate_from_intermediate mydevice` instead to chain up the device certificate to the intermediate certificate.  
```

Comment:
```suggestion
* Run `./certGen.sh create_device_certificate mydevice` to create the new device certificate, which will be chained to the root CA directly.   If the device certificate needs to be signed instead by intermediate CA, run `./certGen.sh create_device_certificate_from_intermediate mydevice` to chain up the device certificate to the intermediate certificate.  
```

Reply:
@nkgami  Please split this into a different paragraph that explains longer chains. There are 2 important things to add in this new documentation for longer chains:
1. If the service (DPS or Hub) was configured with the (top-level) CA, the device is responsible with sending _all_ certificates up to but not including the CA. This is mandatory and allows the service to create a valid chain. 
    - For most TLS stacks (tested on OpenSSL, mbedTLS) this is achieved by creating a PEM certificate store (i.e. concatenating the Intermediates as well as the Device Certificate PEM contents) and passing that to C-SDK. In Windows, all intermediates must be present in the Intermediate Certificate Store and the chain will be automatically built and sent by SChannel.
2. Using a multi-level chain increases the overheads of each TLS connection because the intermediate certificates must be sent every new connection (e.g. for RSA this can add sending several kB of overhead data). It's common that both the client TLS stack and the service have limitations on the TLS Client Hello packet and so long chains (or chains containing several large certificates) may not work.